src/promptflow-evals/promptflow/evals/_common/utils.py

```diff
@@ -5,8 +5,20 @@
     from . import constants
 except ImportError:
     import constants
+
+from typing import List, cast
+
+import nltk
 import numpy as np
 
+try:
+    from nltk.tokenize.nist import NISTTokenizer
+except LookupError:
+    nltk.download("perluniprops")
+    nltk.download("punkt")
+    nltk.download("punkt_tab")
+    from nltk.tokenize.nist import NISTTokenizer
```

Comment:
This feels like hidden dependency logic. Why do we need to download stuff to make this import work, and why can't this be covered by the package's dependency list?

Reply:
I had the same question ealier. Those are nltk data packages, not standard python packages, so we really cannot just add them to the pyproject.toml dependency list. the only to download them is through the python script.